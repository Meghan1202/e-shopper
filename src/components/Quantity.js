/* eslint-disable react/prop-types */
// import PropTypes from 'prop-types';
import React from 'react';

const Quantity = ({ quantity, onIncrement, onDecrement }) => (
  <div>
    <button type="button" onClick={onDecrement}>-</button>
    <p>
      {`${quantity} quantities`}
    </p>
    <button type="button" onClick={onIncrement}> + </button>
  </div>
);

export default Quantity;
