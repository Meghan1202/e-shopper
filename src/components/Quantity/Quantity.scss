@import '../../styles/index.scss';

.quantity__count{
    display: flex;
    align-items: center;
}

button{
    height: 20px;
    width: 20px;
    @include custom-mixin
}