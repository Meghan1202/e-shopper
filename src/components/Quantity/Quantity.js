import React from 'react';
import './Quantity.scss';
import PropTypes from 'prop-types';

const Quantity = ({ count, onIncrement, onDecrement }) => (
  <div className="quantity__count">
    <button type="button" onClick={onDecrement}>-</button>
    <p>
      {`${count} in Basket`}
    </p>
    <button type="button" onClick={onIncrement}> + </button>
  </div>
);

Quantity.propTypes = {
  count: PropTypes.number.isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
};

export default Quantity;
