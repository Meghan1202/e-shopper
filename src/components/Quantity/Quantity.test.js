import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Quantity from './Quantity';

describe(Quantity.name, () => {
  const mockProps = {
    count: 0,
    onIncrement: jest.fn(),
    onDecrement: jest.fn(),
  };
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('should match snapshot', () => {
    const { count, onIncrement, onDecrement } = mockProps;
    const { container } = render(<Quantity
      count={count}
      onIncrement={onIncrement}
      onDecrement={onDecrement}
    />);
    expect(container).toMatchSnapshot();
  });
  test('should display count in basket', () => {
    const { count, onIncrement, onDecrement } = mockProps;
    render(<Quantity
      count={count}
      onIncrement={onIncrement}
      onDecrement={onDecrement}
    />);
    screen.getByText('0 in Basket');
  });
  test('should update the message when increment button is clicked', () => {
    const { count, onIncrement, onDecrement } = mockProps;
    render(<Quantity
      count={count}
      onIncrement={onIncrement}
      onDecrement={onDecrement}
    />);
    const incrementElement = screen.getByText('+');
    fireEvent.click(incrementElement);
    expect(onIncrement).toHaveBeenCalledTimes(1);
  });
  test('should update the message when decrement button is clicked', () => {
    const { count, onIncrement, onDecrement } = mockProps;
    render(<Quantity
      count={count}
      onIncrement={onIncrement}
      onDecrement={onDecrement}
    />);
    const decrementElement = screen.getByText('-');
    fireEvent.click(decrementElement);
    expect(onDecrement).toHaveBeenCalledTimes(1);
  });
  test('should match snapshot', () => {
    const { onIncrement, onDecrement } = mockProps;
    render(<Quantity
      count={50}
      onIncrement={onIncrement}
      onDecrement={onDecrement}
    />);
    screen.getByText('50 in Basket');
  });
});
