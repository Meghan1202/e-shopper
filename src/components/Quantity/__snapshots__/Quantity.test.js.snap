// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quantity should match snapshot 1`] = `
<div>
  <div
    class="quantity__count"
    data-testid="quantity-count"
  >
    <button
      type="button"
    >
      -
    </button>
    <p>
      0 in Basket
    </p>
    <button
      type="button"
    >
       + 
    </button>
  </div>
</div>
`;
